pyspectral/tests/test_rsr_reader.py

```diff
@@ -167,3 +171,279 @@ def test_integral(self, get_rsr_version, download_rsr, load, isfile, exists):
+        check_instrument.return_value = None
+
+        test_rsr = RelativeSpectralResponse('EOS-Aqua', 'modis')
+        test_rsr.set_instrument(self.h5f_aqua_modis)
+        self.assertEqual(test_rsr.instrument, 'modis')
+
+        modis_aqua_filepath = "/mypath/myfilename"
+        test_rsr = RelativeSpectralResponse(filename=modis_aqua_filepath)
+        test_rsr.filename = modis_aqua_filepath
+        test_rsr.platform_name = 'EOS-Aqua'
+        test_rsr.set_instrument(self.h5f_aqua_modis)
+        self.assertEqual(test_rsr.instrument, 'modis')
+
+        viirs_noaa20_filepath = "/mypath/myfilename"
+        test_rsr = RelativeSpectralResponse(filename=viirs_noaa20_filepath)
+        test_rsr.filename = viirs_noaa20_filepath
+
+        test_rsr.set_instrument(self.h5f_noaa20_viirs)
+        self.assertEqual(test_rsr.instrument, 'viirs')
+
+    @patch('pyspectral.rsr_reader.RelativeSpectralResponse.load')
+    @patch('pyspectral.rsr_reader.RelativeSpectralResponse._check_filename_exist')
+    @patch('pyspectral.rsr_reader.RelativeSpectralResponse._get_filename')
+    @patch('pyspectral.rsr_reader.RelativeSpectralResponse._check_instrument')
+    def test_set_band_names(self, check_instrument, get_filename, check_filename_exist, load):
+        """Test setting the band names."""
+
+        load.return_value = None
+        check_filename_exist.return_value = None
+        get_filename.return_value = None
+        check_instrument.return_value = None
+
+        test_rsr = RelativeSpectralResponse('EOS-Aqua', 'modis')
+        test_rsr.set_band_names(self.h5f_aqua_modis)
+        expected = self.modis_bandnames
+        self.assertCountEqual(test_rsr.band_names, expected)
+
+    @patch('pyspectral.rsr_reader.RelativeSpectralResponse.load')
+    @patch('pyspectral.rsr_reader.RelativeSpectralResponse._check_filename_exist')
+    @patch('pyspectral.rsr_reader.RelativeSpectralResponse._get_filename')
+    @patch('pyspectral.rsr_reader.RelativeSpectralResponse._check_instrument')
+    def test_set_band_central_wavelength_per_detector(self,
+                                                      check_instrument, get_filename,
+                                                      check_filename_exist, load):
+        """Test setting the band specific central wavelength for a detector."""
+
+        load.return_value = None
+        check_filename_exist.return_value = None
+        get_filename.return_value = None
+        check_instrument.return_value = None
+
+        test_rsr = RelativeSpectralResponse('NOAA-20', 'viirs')
+
+        h5f = {}
+        for name in self.viirs_bandnames:
+            h5f[name] = {}
+            for det_name in self.viirs_detector_names:
+                h5f[name][det_name] = self.viirs_rsr_data
+
+        test_rsr.rsr['M1'] = {'det-1': {}}
+        test_rsr.set_band_central_wavelength_per_detector(h5f, 'M1', 'det-1')
+        self.assertTrue('central_wavelength' in test_rsr.rsr['M1']['det-1'])
```

Comment:
You could use `assertIn` here to make deepcode happier: https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertIn

Right?

Reply:
Oh, yes, right! thx!